pkg/distributor/distributor.go

```diff
@@ -594,36 +601,40 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	const maxExpectedReplicationSet = 5 // typical replication factor 3 plus one for inactive plus one for luck
 	var descs [maxExpectedReplicationSet]ring.InstanceDesc
 
-	tracker := pushTracker{
+	kafkaTracker := pushTracker{
```

Comment:
The API behaviour is exactly the same. The only benefit of the separate trackers is that we can monitor when all the ingesters complete vs when all the kafka writes complete. Before, the tracker only fired "done" once both sets of writes had completed and there was no way to tell which one was faster.

Initially I did this so we could create a histogram from the timings, but I've since replaced it with Spans. We do get subspans from the ingesters once we hit their Push endpoint, but there are no sub-spans from writing to Kafka which is what I wanted to amend here.